sch_fq.c

```diff
@@ -479,15 +655,62 @@ static int fq_enqueue(struct sk_buff *skb, struct Qdisc *sch,
 	f->qlen++;
 	qdisc_qstats_backlog_inc(sch, skb);
 	if (fq_flow_is_detached(f)) {
-		fq_flow_add_tail(&q->new_flows, f);
+
+	    fq_flow_add_tail(&q->new_flows, f);
+
+
 		if (time_after(jiffies, f->age + q->flow_refill_delay))
 			f->credit = max_t(u32, f->credit, q->quantum);
 		q->inactive_flows--;
 	}
 
 	/* Note: this overwrites f->age */
+
+        //printk("hash of flow value : %u \n ", (f->socket_hash & q->orphan_mask) );
+
+        /*printk("skb get hash value  : %u \n ", skb_get_hash(skb));
+
+        unsigned long pHash = skb_get_hash(skb) & q->orphan_mask;
+
+        printk("pHash value  : %lu \n ", pHash);*/
+
 	flow_queue_add(f, skb);
 
+	/*
+
+	setting the barrier bits
+	there are 100000 barriers for the co-flow
+
+	individual bits are set by two variabes
+
+	time to send being modified
+
+	*/
+
+	if(f->socket_hash == pFlowid[0])
+	{
+	barrier[barriercounter_flow[0]] = barrier[barriercounter_flow[0]] | 1 << 0;
+	
+	printk("pflow id 0  is   : %d \n ", pFlowid[0]);
+
+	fq_skb_cb(skb)->time_to_send = ktime_get_ns() + 10000000;
```

Comment:
Why `1000000`? Where is this coming from?
Don't use magic numbers within the code. If you have to fix a parameter value, first `#define` it with a suitable name at the top of the file, e.g., `#define DELAY_BOUND_NS 1000000` to indicate that the number is a delay bound, and its unit is nanoseconds. Note it is very helpful to also state the unit of the value in the name of the defined variable. And then use `time_to_send = ktime_get_ns() + DELAY_BOUND_NS`